tests/suites/test_suite_cipher.function

```diff
@@ -119,7 +119,8 @@ static int cipher_reset_key( mbedtls_cipher_context_t *ctx, int cipher_id,
     mbedtls_cipher_free( ctx );
     mbedtls_cipher_init( ctx );
 
-#if !defined(MBEDTLS_USE_PSA_CRYPTO)
+#if !defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_DEPRECATED_REMOVED) || \
+    !defined(MBEDTLS_TEST_DEPRECATED)
```

Comment:
I think `DEPRECATED_REMOVED` is redundant with `TEST_DEPRECATED`, so the guard should be `#if !defined(MBEDTLS_USE_PSA_CRYPO) || !defined(MBEDTLS_TEST_DEPRECATED)`. That is, in plain English the condition for being able to use `cipher_setup_psa()` is "use-PSA is enabled, and we're allowed to test deprecated functions".